tests/system/src/test/java/test/com/sun/marlin/DashedRectTest.java

```diff
@@ -0,0 +1,217 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.com.sun.marlin;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.collections.ObservableList;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.scene.SnapshotParameters;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.ClosePath;
+import javafx.scene.shape.HLineTo;
+import javafx.scene.shape.MoveTo;
+import javafx.scene.shape.Path;
+import javafx.scene.shape.VLineTo;
+import javafx.stage.Stage;
+
+import junit.framework.AssertionFailedError;
+import org.junit.AfterClass;
+import static org.junit.Assert.assertEquals;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static test.util.Util.TIMEOUT;
+
+/**
+ * Simple Dashed Rect rendering test
+ *
+ * @test
+ * @summary verify that dashed rectangle is properly rasterized
+ * @bug 8202743
+ */
+public class DashedRectTest {
+
+    static final int BLUE_PIXEL = 0xff0000ff;
+
+    static final File OUTPUT_DIR = new File(".");
+
+    final static double DASH_LEN = 3.0;
+    final static double DASH_PH = 5000.0;
+
+    final static int MAX = 100;
+
+    // Used to launch the application before running any test
+    private static final CountDownLatch launchLatch = new CountDownLatch(1);
+
+    // Singleton Application instance
+    static MyApp myApp;
+
+    // Application class. An instance is created and initialized before running
+    // the first test, and it lives through the execution of all tests.
+    public static class MyApp extends Application {
+
+        Stage stage = null;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            DashedRectTest.myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            this.stage = primaryStage;
+
+            // Corrupt Marlin Dasher.dash cached array:
+            final Path path1 = new Path();
+            path1.getElements().addAll(
+                    new MoveTo(20, 20),
+                    new HLineTo(70),
+                    new VLineTo(70),
+                    new HLineTo(20),
+                    new ClosePath()
+            );
```

Comment:
Minor suggestion: Normally we would populate the content of the scene graph in the test method and not in the Application start method, which is called from the static `@BeforeClass` method.

Since you only have a single test method, it isn't crucial to fix this...think of it more like a best practice. It's up to you whether to make this change.

Reply:
fixed